packages/block-editor/src/components/rich-text/index.js

```diff
@@ -1176,51 +1173,57 @@ RichText.defaultProps = {
 
 const RichTextContainer = compose( [
 	withInstanceId,
-	withBlockEditContext( ( context, ownProps ) => {
-		// When explicitly set as not selected, do nothing.
-		if ( ownProps.isSelected === false ) {
```

Comment:
It exists for backward compatibility in the case where blocks pass `isSelected` explicitly. It should be tested with Gallery and Image blocks to ensure they continue to work properly.

Reply:
It looks like it is no longer necessary for core blocks so I would be fine removing it altogether. We should clean up Gallery and Image blocks:
- https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/image/edit.js#L702
- https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/gallery/gallery-image.js#L146

Reply:
Thanks for checking this! I'll have a look.

Reply:
@gziolo I removed all this code in 5a69d41! Thanks!

Reply:
Oh wait, looks like we still need the state there to remove the toolbar when clicking on the image. This is because we do not _yet_ remove rich text selection state on `blur`. We're one step closer to this with this PR though.

Reply:
Updated in cbf86be. Let's allow it to be set again for now, and work on it further later, in order to not make the PR more complex to review.